autogpt/configurator.py

```diff
@@ -82,10 +82,22 @@ def create_config(
 
     if gpt3only:
```

Comment:
IMO, as a user:

Command line arguments should override `.env` the variables they are meant to override but not do too much [magic](https://en.wikipedia.org/wiki/Magic_(programming)) by fixing incompatibilities with other `.env ` variables or swapping out other variables. Incompatibilities should be made obvious for me to fix.

The "magic " may have short-term benefits," but it can eventually hide bugs that may be hard to spot and fix.

So, if I pass `--gpt3only` on the command line, I expect the following variables to react as follows:
- smart_llm_model = GPT-3.5-turbo, ignore .env file
- fast_llm_model = GPT-3.5-turbo, ignore .env file
- smart_token_limit: Use what's in .env (SMART_LLM_TOKEN_LIMIT). If it's incompatible with `--gpt3only`, the API should raise a clear, appropriate error, and I'll update .`env` file.
- fast_token_limit: Use what's in .env (FAST_LLM_TOKEN_LIMIT). If it's incompatible with `--gpt3only`, if incompatible, the API should fail gracefully, and I'll update the `.env`
- azure_model_to_deployment_id_map["smart_llm_model_deployment_id"] = I expect the gpt3only ID to be used because the smart_llm_model has been set to GPT-3.5-turbo above, not because the system will swap in the env.FAST_LLM_MODEL. 
- azure_model_to_deployment_id_map["smart_llm_model_deployment_id"] 

Similarly, if I pass `--gpt4only`, I expect the system to act transparently by ignoring the config file and setting the chosen LLM model to GPT-4 (not what's in my config file). For `azure_model_to_deployment_id_map`, it'd return GPT4 values for both smart and fast models because it's set smart and fast models to GPT-4. It'd not swap the key it pulls from, which depends on a .env variable.